crates/signer-local/src/lib.rs

```diff
@@ -33,57 +33,58 @@ pub use yubihsm;
 #[cfg(feature = "mnemonic")]
 pub use coins_bip39;
 
-/// A wallet instantiated with a locally stored private key
-pub type LocalWallet = Wallet<k256::ecdsa::SigningKey>;
+/// A signer instantiated with a locally stored private key.
+pub type PrivateKeySigner = LocalSigner<SigningKey>;
 
-/// A wallet instantiated with a YubiHSM
+/// A signer instantiated with a YubiHSM.
 #[cfg(feature = "yubihsm")]
-pub type YubiWallet = Wallet<yubihsm::ecdsa::Signer<k256::Secp256k1>>;
+pub type YubiSigner = LocalSigner<yubihsm::ecdsa::Signer<k256::Secp256k1>>;
 
 /// An Ethereum private-public key pair which can be used for signing messages.
 ///
 /// # Examples
 ///
 /// ## Signing and Verifying a message
 ///
-/// The wallet can be used to produce ECDSA [`Signature`] objects, which can be
+/// The signer can be used to produce ECDSA [`Signature`] objects, which can be
 /// then verified. Note that this uses
 /// [`eip191_hash_message`](alloy_primitives::eip191_hash_message) under the hood which will
 /// prefix the message being hashed with the `Ethereum Signed Message` domain separator.
 ///
 /// ```
 /// use alloy_signer::{Signer, SignerSync};
+/// use alloy_signer_local::PrivateKeySigner;
 ///
-/// let wallet = alloy_signer_wallet::LocalWallet::random();
+/// let signer = PrivateKeySigner::random();
 ///
-/// // Optionally, the wallet's chain id can be set, in order to use EIP-155
+/// // Optionally, the signers' chain id can be set, in order to use EIP-155
 /// // replay protection with different chains
-/// let wallet = wallet.with_chain_id(Some(1337));
+/// let signer = signer.with_chain_id(Some(1337));
 ///
-/// // The wallet can be used to sign messages
+/// // The signer can be used to sign messages
 /// let message = b"hello";
-/// let signature = wallet.sign_message_sync(message)?;
-/// assert_eq!(signature.recover_address_from_msg(&message[..]).unwrap(), wallet.address());
+/// let signature = signer.sign_message_sync(message)?;
+/// assert_eq!(signature.recover_address_from_msg(&message[..]).unwrap(), signer.address());
 ///
-/// // LocalWallet is cloneable:
-/// let wallet_clone = wallet.clone();
-/// let signature2 = wallet_clone.sign_message_sync(message)?;
+/// // LocalSigner is cloneable:
+/// let signer_clone = signer.clone();
+/// let signature2 = signer_clone.sign_message_sync(message)?;
 /// assert_eq!(signature, signature2);
 /// # Ok::<_, Box<dyn std::error::Error>>(())
 /// ```
 #[derive(Clone)]
-pub struct Wallet<D> {
-    /// The wallet's private key.
-    pub(crate) signer: D,
-    /// The wallet's address.
+pub struct LocalSigner<C> {
+    /// The signers' credential.
+    pub(crate) credential: C,
```

Comment:
Referring to inner `private_key` as `credential` for flexibility, open to changing it